In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
pd.set_option('display.max_columns', None)
pd.options.display.float_format = '{:.2f}'.format

In [12]:
df_dirty = pd.read_csv(r'C:\Users\rashid\Desktop\funnel_analysis\data\merged\merged_data.csv')
df = df_dirty.copy()
print(df.shape)
df.head()

(8421434, 9)


,event_time,event_type,product_id,category_id,category_code,brand,price,user_id,user_session
0,2020-01-01 00:00:00 UTC,view,5809910,1602943681873052386,NaN,grattol,5.24,595414620,4adb70bb-edbd-4981-b60f-a05bfd32683a
1,2020-01-01 00:00:09 UTC,view,5812943,1487580012121948301,NaN,kinetics,3.97,595414640,c8c5205d-be43-4f1d-aa56-4828b8151c8a
2,2020-01-01 00:00:19 UTC,view,5798924,1783999068867920626,NaN,zinger,3.97,595412617,46a5010f-bd69-4fbe-a00d-bb17aa7b46f3
3,2020-01-01 00:00:24 UTC,view,5793052,1487580005754995573,NaN,NaN,4.92,420652863,546f6af3-a517-4752-a98b-80c4c5860711
4,2020-01-01 00:00:25 UTC,view,5899926,2115334439910245200,NaN,NaN,3.92,484071203,cff70ddf-529e-4b0c-a4fc-f43a749c0acb


# Dataset Metaməlumatı (Metadata)

Bu dataset istifadəçilərin e-ticarət platformasındakı davranışlarını və əməliyyatlarını əks etdirir.

## Sütunların Təsviri (Data Dictionary)

| Sütun Adı | Məlumat Tipi (Dtype) | Təsviri / İzahı | Misal Dəyər |
| :--- | :--- | :--- | :--- |
| **`event_time`** | `str` / `datetime` | Əməliyyatın və ya hadisənin baş verdiyi dəqiq tarix və vaxt (UTC formatında). | `2020-01-01 00:00:00+00:00` |
| **`event_type`** | `str` | İstifadəçi fəaliyyətinin növü (məsələn: `view` - baxış, `cart` - səbətə atma, `purchase` - alıcılıq). | `view` |
| **`product_id`** | `int64` | Məhsulun unikal identifikasiya nömrəsi (ID). | `5809910` |
| **`category_id`** | `int64` | Məhsulun aid olduğu kateqoriyanın unikal ID-si. | `1602943681873052386` |
| **`category_code`** | `str` | Kateqoriyanın oxunabilən adı/kodu (kateqoriya iyerarxiyası). Çox vaxt boş (`NaN`) ola bilər. | `appliances.environment.air_heater` / `NaN` |
| **`brand`** | `str` | Məhsulun brend/marka adı. | `grattol` |
| **`price`** | `float64` | Məhsulun qiyməti (valyuta vahidində, float tipli). | `5.24` |
| **`user_id`** | `int64` | Istifadəçinin unikal ID nömrəsi. | `595414620` |
| **`user_session`** | `str` | İstifadəçinin bir sessiyasına aid olan unikal kod (UUID). | `4adb70bb-edbd-4981-b60f-a05bfd32683a` |

---

## Genel Dataset Xülasəsi

- **Ümumi Sətir Sayı:** 8,421,434
- **Ümumi Sütun Sayı:** 9
- **Yaddaş İstifadəsi:** ~578.3 MB

In [4]:
df.info(show_counts=True)

<class 'pandas.DataFrame'>
RangeIndex: 8421434 entries, 0 to 8421433
Data columns (total 9 columns):
 #   Column         Non-Null Count    Dtype  
---  ------         --------------    -----  
 0   event_time     8421434 non-null  str    
 1   event_type     8421434 non-null  str    
 2   product_id     8421434 non-null  int64  
 3   category_id    8421434 non-null  int64  
 4   category_code  151904 non-null   str    
 5   brand          4819896 non-null  str    
 6   price          8421434 non-null  float64
 7   user_id        8421434 non-null  int64  
 8   user_session   8419065 non-null  str    
dtypes: float64(1), int64(3), str(5)
memory usage: 1.1 GB


In [6]:
df['event_time'] = pd.to_datetime(df['event_time'], errors='coerce')

In [7]:
df.isna().sum()

event_time             0
event_type             0
product_id             0
category_id            0
category_code    8269530
brand            3601538
price                  0
user_id                0
user_session        2369
dtype: int64

In [ ]:
df.duplicated().sum()

np.int64(465390)

In [9]:
df = df.drop_duplicates()

In [11]:
df.to_parquet(r'C:\Users\rashid\Desktop\funnel_analysis\data\processed\cleaned_data.parquet', index=False, compression='snappy')
